src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -255,6 +257,27 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
         appModel.Resources.AddRange(sideCars);
     }
 
+    // This method resolves the application's endpoint and the protocol that the dapr side car will use.
+    // It depends on DaprSidecarOptions.AppProtocol and DaprSidecarOptions.AppEndpoint.
+    // - If both are null default to 'http' for both.
+    // - If AppProtocol is not null try to get an endpoint with the name of the protocol.
+    // - if AppEndpoint is not null try to use the scheme as the protocol.
+    // - if both are not null just use both options.
+    static (EndpointReference appEndpoint, string protocol)? GetEndpointReference(DaprSidecarOptions? sidecarOptions, IResource resource)
+    {
+        if (resource is IResourceWithEndpoints resourceWithEndpoints)
+        {
+            return (sidecarOptions, resourceWithEndpoints) switch{
+                (var p0, var p1) when p0 == null || (p0.AppProtocol == null && p0.AppEndpoint == null) => (p1.GetEndpoint("http"), "http"),
```

Comment:
❗  Equality operator can be overloaded which can load to hard to track bugs.
```suggestion
                (var p0, var p1) when p0 is null || (p0.AppProtocol is null && p0.AppEndpoint is null) => (p1.GetEndpoint("http"), "http"),
```

ditto other places

Reply:
done